packages/bot-handler-avs/src/AVSHandler.ts

```diff
@@ -0,0 +1,200 @@
+    }
+
+    const avsResult = this.wCall!.recvMsg(
+      this.wUser!,
+      callContent,
+      callContent.length,
+      new Date().getSeconds(),
+      new Date(callMessage.timestamp).getSeconds(),
+      callMessage.conversation,
+      callMessage.from,
+      callMessage.fromClientId,
+    );
+
+    if (avsResult !== 0) {
+      throw new Error(`recvMsg failed with code "${avsResult}"`);
+    }
+
+    this.answerCall(callMessage);
+  }
+
+  startCall(conversationId: string, conversationType: CONV_TYPE, callType: CALL_TYPE): void {
+    this.wCall!.start(this.wUser!, conversationId, callType, conversationType, 0);
+  }
+
+  private answerCall(call: CallMessage) {
+    this.wCall!.answer(this.wUser!, call.conversation, CALL_TYPE.NORMAL, 0);
+  }
+
+  private configureCallingApi(wCall: Wcall): Wcall {
+    wCall.setLogHandler((level: LOG_LEVEL, message: string) => {
+      console.info(`${level}: ${message}`);
+    });
+
+    const avsEnv = AVS_ENV.DEFAULT;
+    wCall.init(avsEnv);
+    wCall.setUserMediaHandler(this.mediaHandler);
+    wCall.setMediaStreamHandler(() => {});
+    setInterval(() => wCall.poll(), 500);
+    return wCall;
+  }
+
+  private readonly mediaHandler = async (): Promise<MediaStream> => {
+    return this.mediaStream;
+  };
+
+  private readonly createWUser = (wCall: Wcall, selfUserId: string, selfClientId: string): number => {
+    /* cspell:disable */
+    const wUser = wCall.create(
+      selfUserId,
+      selfClientId,
+      () => {},
+      this.onSendCallMessage,
+      this.sendSFTRequest,
+      this.onIncomingCall,
+      () => {},
+      () => {},
+      () => {},
+      this.callClosed,
+      this.callConfigRequestHandler,
+      () => 0,
+      () => {},
+      () => {},
+    );
+
+    wCall.setParticipantChangedHandler(wUser, () => {});
+    return wUser;
+  };
+
+  private callConfigRequestHandler(convid: string, metrics_json: string, arg: number) {
+    this.account!.service!.account.getCallConfig()
+      .then(config => {
+        this.wCall!.configUpdate(this.wUser!, 0, JSON.stringify(config));
+      })
+      .catch(console.error);
+
+    return 0;
+  }
+
+  private readonly sendSFTRequest = (
+    context: number,
+    url: string,
+    data: string,
+    dataLength: number,
+    _: number,
+  ): number => {
+    void (async () => {
+      try {
+        const response = await axios.post(url, data);
+        const {status, data: axiosData} = response;
+        const jsonData = JSON.stringify(axiosData);
+        this.wCall!.sftResp(this.wUser!, status, jsonData, jsonData.length, context);
+      } catch (error) {
+        console.info('Failed to send SFT request', error);
+      }
+    })();
+    return 0;
+  };
+
+  private readonly onSendCallMessage = (
+    ctx: number,
+    conversationId: string,
+    selfUserId: string,
+    selfClientId: string,
+    userid_dest: string | undefined,
+    clientid_dest: string | undefined,
+    data: string,
```

Comment:
A required parameter should not follow an optional parameter

Reply:
This is a given from the AVS API I think

Reply:
Ah okay